--- ./src/playaudio.cpp.orig	2008-04-21 00:35:49.000000000 +0200
+++ ./src/playaudio.cpp	2008-04-21 00:36:16.000000000 +0200
@@ -38,7 +38,7 @@
   const uint8_t *d=(const uint8_t*)data;
 
   while (len>0) {
-    int16_t samples[MIN_BUFFER_SAMPLES >? avcc->frame_size];
+    int16_t samples[MIN_BUFFER_SAMPLES > avcc->frame_size ? MIN_BUFFER_SAMPLES : avcc->frame_size ];
     int frame_size;
 
     int bytesDecoded=avcodec_decode_audio(avcc,samples,&frame_size,(uint8_t*)d,len);
--- ./src/mpegmuxer.cpp.orig	2008-04-21 17:10:55.000000000 +0200
+++ ./src/mpegmuxer.cpp	2008-04-21 17:11:31.000000000 +0200
@@ -23,6 +23,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <assert.h>
+#include <cstdlib>
 #include "mpegmuxer.h"
 
 #ifndef O_BINARY
--- ./src/mpgfile.cpp.orig	2008-04-21 17:12:56.000000000 +0200
+++ ./src/mpgfile.cpp	2008-04-21 17:13:51.000000000 +0200
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <math.h>
+#include <cstdlib>
 
 #include <list>
 #include <utility>
