Description: Do not access argv beyond last item
Author: Bernhard Übelacker <bernhardu@vr-web.de>
Bug-Debian: https://bugs.debian.org/715843

--- dvbcut-0.5.4+svn178.orig/src/main.cpp
+++ dvbcut-0.5.4+svn178/src/main.cpp
@@ -108,18 +108,18 @@ main(int argc, char *argv[]) {
         generateidx = true;
       else if (strncmp(argv[i], "-voracious", n) == 0)
 	dvbcut::cache_friendly = false;
-      else if (strncmp(argv[i], "-idx", n) == 0 && ++i < argc)
-        idxfilename = argv[i];
-      else if (strncmp(argv[i], "-exp", n) == 0 && ++i < argc)
-        expfilename = argv[i];
-      else if (strncmp(argv[i], "-format", n) == 0 && ++i < argc)
-        exportformat = atoi(argv[i]);
-      else if (strncmp(argv[i], "-automarker", n) == 0 && ++i < argc) {
-        int bofeof = atoi(argv[i]);
+      else if (strncmp(argv[i], "-idx", n) == 0 && (i+1) < argc)
+        idxfilename = argv[++i];
+      else if (strncmp(argv[i], "-exp", n) == 0 && (i+1) < argc)
+        expfilename = argv[++i];
+      else if (strncmp(argv[i], "-format", n) == 0 && (i+1) < argc)
+        exportformat = atoi(argv[++i]);
+      else if (strncmp(argv[i], "-automarker", n) == 0 && (i+1) < argc) {
+        int bofeof = atoi(argv[++i]);
         start_bof = (bofeof&1)==1;
         stop_eof  = (bofeof&2)==2;
-      } else if (strncmp(argv[i], "-cut", n) == 0 && ++i < argc) {
-        char *pch = strtok(argv[i],",-|;");
+      } else if (strncmp(argv[i], "-cut", n) == 0 && (i+1) < argc) {
+        char *pch = strtok(argv[++i],",-|;");
         while(pch) {
           if(strlen(pch))
             cutlist.push_back((std::string)pch);    
