Description: mplayer2 uses different indices for audio streams
Author: Bernhard Übelacker <bernhardu@vr-web.de>
Bug-Debian: https://bugs.debian.org/753369

--- dvbcut-0.5.4+svn178.orig/src/dvbcut.cpp
+++ dvbcut-0.5.4+svn178/src/dvbcut.cpp
@@ -1182,7 +1182,7 @@ void dvbcut::playPlay()
   arguments << "-geometry" << QString().sprintf("%dx%d+0+0",int(ui->imagedisplay->width()),int(ui->imagedisplay->height()));
 
   if (currentaudiotrack>=0 && currentaudiotrack<mpg->getaudiostreams()) {
-    arguments << "-aid" << QString().sprintf("0x%x",int(mpg->mplayeraudioid(currentaudiotrack)));
+    arguments << "-aid" << QString().sprintf("0x%x",int(currentaudiotrack));
     }
     
   // for now, pass all filenames from the current one up to the last one
